app/views/onboarding/Onboarding4.js

```diff
@@ -25,10 +27,11 @@ const Onboarding = (props) => {
   } = useAssets();
 
   const handleOnPressNext = () => {
-    const nextScreen = isGPS
-      ? 'OnboardingPermissions'
-      : 'EnableExposureNotifications';
-    props.navigation.replace(nextScreen);
+    if (isGPS) {
+      props.navigation.replace('OnboardingPermissions', { step: isiOS ? 'notification' : 'location' });
```

Comment:
is there a way we can do this without manually passing data through react-navigation? could the screen itself handle its conditional logic?